apis/run/v1alpha1/conversion.go

```diff
@@ -0,0 +1,149 @@
+	corednsContainerImageName = "coredns"
+	pauseContainerImageName   = "pause"
+)
+
+func (spoke *TanzuKubernetesRelease) ConvertTo(hubRaw conversion.Hub) error {
+	hub := hubRaw.(*v1alpha3.TanzuKubernetesRelease)
+
+	if err := autoConvert_v1alpha1_TanzuKubernetesRelease_To_v1alpha3_TanzuKubernetesRelease(spoke, hub, nil); err != nil {
+		return nil
+	}
+
+	// Some Hub types not present in spoke, and might be stored in spoke annotations.
+	// Restore data stored in spoke's annotations.
+	restoredHub := &v1alpha3.TanzuKubernetesRelease{}
+	if ok, err := utilconversion.UnmarshalData(spoke, restoredHub); err != nil {
+		return err
+	} else if ok {
+		// OK means there is data in annotations that need to be restored.
+		hub.Spec.BootstrapPackages = restoredHub.Spec.BootstrapPackages
+		hub.Spec.OSImages = restoredHub.Spec.OSImages
+	}
+
+	// Store the entire spoke in Hub's annotations to prevent loss of incompatible spoke types.
+	if err := utilconversion.MarshalData(spoke, hub); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (spoke *TanzuKubernetesRelease) ConvertFrom(hubRaw conversion.Hub) error {
+	hub := hubRaw.(*v1alpha3.TanzuKubernetesRelease)
+
+	if err := autoConvert_v1alpha3_TanzuKubernetesRelease_To_v1alpha1_TanzuKubernetesRelease(hub, spoke, nil); err != nil {
+		return err
+	}
+
+	// From hub, get missing spoke fields from annotations.
+	restored := &TanzuKubernetesRelease{}
+	if ok, err := utilconversion.UnmarshalData(hub, restored); err != nil {
+		return err
+	} else if ok {
+		// Annotations contain data, restore the relevant ones.
+		spoke.Spec.NodeImageRef = restored.Spec.NodeImageRef
+		if restored.Spec.Images != nil {
+			for _, image := range restored.Spec.Images {
+				if image.Name != etcdContainerImageName && image.Name != pauseContainerImageName && image.Name != corednsContainerImageName {
+					// Do not copy over etcd, pause and coredns as those are compatible and handled in the autoconvert.
+					spoke.Spec.Images = append(spoke.Spec.Images, image)
+				}
+			}
+		}
+	}
+
+	// Store hub's missing fields in spoke's annotations.
+	if err := utilconversion.MarshalData(hub, spoke); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Convert_v1alpha1_TanzuKubernetesReleaseSpec_To_v1alpha3_TanzuKubernetesReleaseSpec  will convert the compatible types in TanzuKubernetesReleaseSpec v1alpha1 to v1alpha3 equivalent.
+func Convert_v1alpha1_TanzuKubernetesReleaseSpec_To_v1alpha3_TanzuKubernetesReleaseSpec(in *TanzuKubernetesReleaseSpec, out *v1alpha3.TanzuKubernetesReleaseSpec, s apiconversion.Scope) error {
+	out.Kubernetes.Version = in.KubernetesVersion
+	out.Kubernetes.ImageRepository = in.Repository
+
+	// Transform the container images.
+	for index, image := range in.Images {
+		switch in.Images[index].Name {
+		case etcdContainerImageName:
+			out.Kubernetes.Etcd = &v1alpha3.ContainerImageInfo{
+				ImageRepository: image.Repository,
+				ImageTag:        image.Tag,
+			}
+		case pauseContainerImageName:
```

Comment:
I see in the current v1alpha1 TKR there are a bunch of images available for `pause` image (as shown below). Hope there isn't any issue copying the last entry of `pause` image in the v1alpha1 TKR list to v1alpha3 TKR's `pause` image
```
 - name: pause
    repository: projects-stg.registry.vmware.com/tkg
    tag: "3.5"
  - name: pause
    repository: projects-stg.registry.vmware.com/tkg
    tag: 3.5-windows-amd64
  - name: pause
    repository: projects-stg.registry.vmware.com/tkg
    tag: 3.5-windows-amd64-1903
```

Reply:
Handled in cf38543c53922f7ac1aee651fbe9eb89284e4daa